src/ocnicepost.fd/ocnicepost.F90

```diff
@@ -0,0 +1,577 @@
+  i = 0; j = 0; k = 0
+  do n = 1,nvalid
+     if (trim(outvars(n)%var_remapmethod) == 'bilinear') then
+        if (outvars(n)%var_dimen == 2 .and. allocated(b2d)) then
+           i = i+1; b2d(i) = outvars(n)
+        end if
+        if (outvars(n)%var_dimen == 3 .and. allocated(b3d)) then
+           j = j+1; b3d(j) = outvars(n)
+        end if
+     end if
+     if (trim(outvars(n)%var_remapmethod) == 'conserve' .and. allocated(c2d)) then
+        k = k+1; c2d(k) = outvars(n)
+     end if
+  end do
+
+  ! --------------------------------------------------------
+  ! create packed arrays for mapping
+  ! --------------------------------------------------------
+
+  ! 2D bilin
+  if (allocated(bilin2d)) then
+     call packarrays(trim(input_file), trim(wgtsdir), cosrot, sinrot, b2d, dims=(/nxt,nyt/),       &
+          nflds=nbilin2d, fields=bilin2d)
+
+     if (debug) then
+        write(logunit,'(/,a)')'2D fields mapped bilin, packed field min/max values'
+        do n = 1,nbilin2d
+           write(logunit,'(i6,4(a,a),2g14.4)')n,'  ',trim(b2d(n)%input_var_name),'  ',             &
+                trim(b2d(n)%var_grid),'  ',trim(b2d(n)%var_pair),'  ', trim(b2d(n)%var_pair_grid), &
+                minval(bilin2d(:,n)), maxval(bilin2d(:,n))
+        end do
+        call dumpnc(trim(ftype)//'.bilin2d.nc', 'bilin2d', dims=(/nxt,nyt/), nflds=nbilin2d, field=bilin2d)
+     end if
+  end if
+
+  ! 2D conserv
+  if (allocated(consd2d)) then
+     call packarrays(trim(input_file), trim(wgtsdir), cosrot, sinrot, c2d, dims=(/nxt,nyt/),       &
+          nflds=nconsd2d, fields=consd2d)
+
+     if (debug) then
+        write(logunit,'(a)')'2D fields mapped conserv, packed field min/max values'
+        do n = 1,nconsd2d
+           write(logunit,'(i6,4(a,a),2g14.4)')n,'  ',trim(c2d(n)%input_var_name),'  ',             &
+                trim(c2d(n)%var_grid),'  ',trim(c2d(n)%var_pair),'  ', trim(c2d(n)%var_pair_grid), &
+                minval(consd2d(:,n)), maxval(consd2d(:,n))
+        end do
+        call dumpnc(trim(ftype)//'.consd2d.nc', 'consd2d', dims=(/nxt,nyt/), nflds=nconsd2d, field=consd2d)
+     end if
+  end if
+
+  ! 3D bilin
+  if (allocated(bilin3d))then
+     call packarrays(trim(input_file), trim(wgtsdir), cosrot, sinrot, b3d, dims=(/nxt,nyt,nlevs/), &
+          nflds=nbilin3d, fields=bilin3d)
+
+     if (debug) then
+        write(logunit,'(a)')'3D fields mapped bilin, packed field min/max values'
+        do n = 1,nbilin3d
+           write(logunit,'(i6,4(a,a),2g14.4)')n,'  ',trim(b3d(n)%input_var_name),'  ',             &
+                trim(b3d(n)%var_grid),'  ',trim(b3d(n)%var_pair),'  ', trim(b3d(n)%var_pair_grid), &
+                minval(bilin3d(:,:,n)), maxval(bilin3d(:,:,n))
+        end do
+        call dumpnc(trim(ftype)//'.bilin3d.nc', 'bilin3d', dims=(/nxt,nyt,nlevs/), nk=nlevs,       &
+             nflds=nbilin3d, field=bilin3d)
+     end if
+  end if
+
+  ! --------------------------------------------------------
```

Comment:
The remap application should be moved to it's own module and subroutine(s). Further, the ncio package noted above should be used for the netCDF interfaces.